airflow/models/dagrun.py

```diff
@@ -494,12 +542,13 @@ def verify_integrity(self, session: Session = None):
                 session.add(ti)
 
         try:
-            session.commit()
+            session.flush()
         except IntegrityError as err:
             self.log.info(str(err))
             self.log.info('Hit IntegrityError while creating the TIs for '
                           f'{dag.dag_id} - {self.execution_date}.')
             self.log.info('Doing session rollback.')
+            # TODO[HA]: We probaly need to savepoint this so we can keep the transaction alive.
```

Comment:
On a side not. Should we rename verify_integrity to some different name ? Seems that this method not only verify the integrity of the dagrun and connected task instances but it also creates task instances were they are missing. Maybe "synchronize_task_instances" or something like that.

Reply:
Already had that thought too :) 
https://github.com/apache/airflow/blob/b28d67332bb052441bc2ffd00030d620aac381c8/airflow/jobs/scheduler_job.py#L1565-L1566

Reply:
Any thoughs on that? I think it is a bit misleading with this name.

Reply:
Deferring this -- the behaviour or name of this function hasn't changed in this PR